pyxem/utils/indexation_utils.py

```diff
@@ -38,17 +40,39 @@
 OrientationResult = namedtuple("OrientationResult",
                                "phase_index rotation_matrix match_rate error_hkls total_error scale center_x center_y".split())
 
+#Functions used in correlate_library.
+def Fast_Correlation(image_intensities,int_local,pn_local, **kwargs):
+    return np.sum(np.multiply(image_intensities, int_local)) / \
+                pn_local  # Correlation is the partially normalized dot product
+
+def Normalized_Correlation(N,image_norm,average_image_intensity,image_intensities,int_local, **kwargs):
+    N_star = len(image_intensities)
```

Comment:
variables start with lower cases. Also anything that doesn't depend on the library should be done outside of this function, because we run every library entry against an image, so anything that doesn't change can be calculated once rather the N times (N size of library)